### PR TITLE
Feature/payments/ci

### PR DESCRIPTION
**Pull request by Joseph-Nash**
Merge ci branch to develop
- Update README.md
- Update pom.xml with up to date spring dependencies and remove use of spring-boot-starter-test (we are including submodule spring-test within this that is needed). 
- Add logging interceptor
- Change scope of ApiClientService variable
- Include piwik event information
- Include user bar fragment in base layout

### Type of change

* [] Bug fix
* [x] New feature
* [] Breaking change

### Pull request checklist

* [x] I have added unit tests for new code that I have added
* [x] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -33,6 +33,9 @@
         <common-web-java.version>1.0.0-rc2</common-web-java.version>
         <joda-money.version>1.0.1</joda-money.version>
 
+        <spring-test.version>5.1.3.RELEASE</spring-test.version>
+
+
```
Comment (leahtaylor) [inline pom.xml]: can you remove this empty line please

File: pom.xml [hunk 2]
```diff
@@ -100,6 +98,20 @@
             <scope>test</scope>
         </dependency>
 
+        <!-- Test scope -->
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest</artifactId>
+            <version>2.1</version>
+        </dependency>
```
Comment (leahtaylor) [inline pom.xml]: Could we set the version in the dependency versions section at the top of the pom file as seen in the spring framework dependency below?

e.g. :  <version>${hamcrest-library-version}</version>

And then in the dependency versions section at the top of the file:

 <hamcrest-library-version>1.3</hamcrest-library-version>

File: pom.xml [hunk 3]
```diff
@@ -100,6 +98,20 @@
             <scope>test</scope>
         </dependency>
 
+        <!-- Test scope -->
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest</artifactId>
+            <version>2.1</version>
```
Comment (caismith-ch) [inline pom.xml]: The version number could be added to the properties section above.
`<hamcrest.version>2.1</hamcrest.version>`

Then used by having `<version>${hamcrest.version}</version>`

File: pom.xml (hunk 1 above)
Comment (Joseph-Nash) [reply pom.xml]: Cool, changed this 👍

File: pom.xml (hunk 2 above)
Comment (Joseph-Nash) [reply pom.xml]: 👍

**Pull request merged**